backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -172,16 +167,88 @@ public UploadResponse getUploadSubmission(UUID id)
             .orElseThrow(InvalidBulkTestResultUploadException::new);
 
+    return _client.getSubmission(result.getReportId(), getRSAuthToken().getAccessToken());
+  }
+
+  public TokenResponse getRSAuthToken() {
     Map<String, String> queryParams = new LinkedHashMap<>();
     queryParams.put("scope", scope);
     queryParams.put("grant_type", "client_credentials");
     queryParams.put(
         "client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
     queryParams.put("client_assertion", createDataHubSenderToken(signingKey));
 
-    TokenResponse r = _client.fetchAccessToken(queryParams);
+    return _client.fetchAccessToken(queryParams);
+  }
+
+  private Future<UploadResponse> submitResultsAsFhir(
+      ByteArrayInputStream content, Organization org) {
+    // convert csv to fhir and serialize to json
+    var serializedFhirBundles = fhirConverter.convertToFhirBundles(content, org.getInternalId());
 
-    return _client.getSubmission(result.getReportId(), r.getAccessToken());
+    // build the ndjson request body
+    var ndJson = new StringBuilder();
+    for (String bundle : serializedFhirBundles) {
+      ndJson.append(bundle).append(System.lineSeparator());
+    }
+
+    // send to report stream
+    return CompletableFuture.supplyAsync(
+        () -> {
+          UploadResponse response;
+          try {
+            response =
+                _client.uploadFhir(ndJson.toString().trim(), getRSAuthToken().getAccessToken());
+          } catch (FeignException e) {
+            log.info("RS Fhir API Error " + e.status() + " Response: " + e.contentUTF8());
+            response = parseFeignException(e);
+          }
+          return response;
+        });
+  }
+
+  private Future<UploadResponse> submitResultsAsCsv(byte[] content) {
+    return CompletableFuture.supplyAsync(
+        () -> {
+          UploadResponse response;
+          try {
+            response = _client.uploadCSV(content);
+          } catch (FeignException e) {
+            log.info("RS CSV API Error " + e.status() + " Response: " + e.contentUTF8());
+            response = parseFeignException(e);
+          }
+          return response;
+        });
+  }
+
+  private UploadResponse parseFeignException(FeignException e) {
+    try {
+      return mapper.readValue(e.contentUTF8(), UploadResponse.class);
+    } catch (JsonProcessingException ex) {
+      log.error("Unable to parse Report Stream response.", ex);
+      return null;
+    }
+  }
+
+  private TestResultUpload saveSubmissionToDb(UploadResponse response, Organization org) {
+    TestResultUpload result = null;
+    if (response != null) {
+      var status = UploadResponse.parseStatus(response.getOverallStatus());
+
+      result =
+          new TestResultUpload(
+              response.getId(),
+              status,
+              response.getReportItemCount(),
+              org,
+              response.getWarnings(),
+              response.getErrors());
```

Comment:
I think it would be valuable to also add information about which pipeline it was sent to. I think another ticket should be created after this to take care of that work. 
![image](https://github.com/CDCgov/prime-simplereport/assets/10108172/07dcd6ee-8b6e-4f1e-a92f-96bc6e26e8c0)